### PR TITLE
Reworded a few things and relocated error handling to the course class

### PR DESCRIPTION
**Pull request by BradleyCai**
Relocated error handling to the course class and reworded a few front end things to convey meaning a little better. Solves issue #25 

image
image

### CODE REVIEW

**Comment by emmiegit** [comment]
I think that if both building and room are `TBA` then it should only display once. Otherwise, I think this is a good change. `BCOE TBA` is much more informative than just a blank line.

File: js/Course.js [hunk 1]
```diff
@@ -15,12 +15,12 @@ function Course(quarter, name, nameID, gt, units,
                 days, hour1, min1, hour2, min2, bldg, room) {
 
     //General information
-    this.quarter = quarter; //written as a string in the format of "season####"
+    this.quarter = (quarter === "" ? "None provided" : quarter); //written as a string in the format of "season####"
```
Comment (emmiegit) [inline js/Course.js:18]: Why `===`? Can you be sure that all JavaScript interpreters see the empty string as the same object globally?

**Comment by BradleyCai** [comment]
@ammongit I'll add that right now

File: js/UcrBuildings.js [hunk 2]
```diff
@@ -153,7 +153,7 @@ function UcrBuildings() {
         var url = "http://campusmap.ucr.edu/imap/index.html?loc=" + bldg;
 
         if (name === undefined)
```
Comment (ammongit) [inline js/UcrBuildings.js]: Some interpreters return `"undefined"` as opposed to `undefined`.

File: js/Course.js (hunk 1 above)
Comment (BradleyCai) [inline js/Course.js:18]: CourseParser.js assigns "" to quarter if the regex for finding the quarter fails. We can be pretty sure that they will both be strings.

**Comment by emmiegit** [comment]
Looks good.

File: js/Course.js (hunk 1 above)
Comment (emmiegit) [inline js/Course.js:18]: But `===` tests for absolute equality. If you had two empty strings that occupied different locations in memory, then `quarter === ""` would be `false`.

Comment (BradleyCai) [inline js/Course.js:18]: It doesn't check for absolute equality. == checks for equality between any two types and === checks for equality for specifically the same type. 
image

**Pull request merged**